src/Temporalio/Client/WorkerTaskReachability.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Contains information about the reachability of some Build IDs.
+    /// </summary>
+    /// <param name="BuildIdReachability">Maps Build IDs to information about their reachability.</param>
+    public record WorkerTaskReachability(IReadOnlyDictionary<string, BuildIdReachability> BuildIdReachability)
+    {
+        /// <summary>
+        /// Convert from proto.
+        /// </summary>
+        /// <param name="proto">Proto.</param>
+        /// <returns>Converted value.</returns>
+        internal static WorkerTaskReachability FromProto(Api.WorkflowService.V1.GetWorkerTaskReachabilityResponse proto)
+        {
+            var idMap = proto.BuildIdReachability.ToDictionary(
+                bid => bid.BuildId,
+                bid =>
```

Comment:
A `internal static BuildIdReachability FromProto(...` in `BuildIdReachability` may be a bit clearer and more consistent